ast/compile.go

```diff
@@ -1781,6 +1804,7 @@ func getComprehensionIndex(arity func(Ref) int, candidates VarSet, expr *Expr) *
 	unsafe := body.Vars(SafetyCheckVisitorParams).Diff(outputs).Diff(ReservedVars)
 
 	if len(unsafe) > 0 {
+		dbg.Printf("%s: unsafe vars: %v", expr, unsafe)
```

Comment:
I'd rather see the file location (`<name>:<line>`) as opposed to the expression here. When looking at the debug logs, it'll be easier to identify the messages of interest using the source location. The unsafe set is good.